pkg/appcommon/harness.go

```diff
@@ -133,15 +133,17 @@ func New(cfg Config, reg prometheus.Registerer, metricPrefix string, tracer open
 		authMiddleware = middleware.HTTPFakeAuth{}
 	}
 
-	requestLimitsMiddleware := middleware.NewRequestLimitsMiddleware(cfg.ServerConfig.HTTPMaxRequestSizeLimit)
-
 	// Middlewares will be wrapped in order
 	middlewares := []middleware.Interface{
 		tracerMiddleware,
 		instrumentMiddleware,
 		authMiddleware,
 		logMiddleware,
-		requestLimitsMiddleware,
+	}
+
+	if (cfg.ServerConfig.HTTPMaxRequestSizeLimit > 0) {
```

Comment:
Why not just change the middleware if the limit is 0?
```
func (l RequestLimits) Wrap(next http.Handler) http.Handler {
        if l.maxRequestBodySize == 0 {
            // skip middleware reading request
            return next;
        }
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		log, _ := spanlogger.New(r.Context(), "middleware.RequestLimits.Wrap")
		defer log.Span.Finish()

		reader := io.LimitReader(r.Body, int64(l.maxRequestBodySize)+1)
```

Reply:
```
flags.Int64Var(&cfg.HTTPMaxRequestSizeLimit, prefix+"server.http-max-req-size-limit", defaultHTTPRequestSizeLimit, "HTTP max request body size limit in bytes")
```
The default is 10MiB - you'd need to explicitly set it to 0 to disable request reading, but that would leave us open to people sending 1GiB payloads, and taking whatever proxy doesn't enforce the limit offline.

Reply:
no strong opinion, only that not inclduing the middleware if it's not wanted seems a bit more straightforward pattern

Reply:
> The default is 10MiB - you'd need to explicitly set it to 0 to disable request reading, but that would leave us open to people sending 1GiB payloads, and taking whatever proxy doesn't enforce the limit offline.

I think a simple proxy shouldn't be reading requests at all, in which case it won't hurt it. OTLP gw should just proxy to mimir/loki, who use the limiter as well. So the end result would be simply that body is not read and limit is not checked twice. Does that make sense?